monitoring/poison_message_queue_finder.py

```diff
@@ -62,5 +67,42 @@ def main():
             print(json.dumps(json_to_log))
 
 
+def get_connection_stats():
+
+    response = requests.get(f"http://{Config.RABBITMQ_HOST}:{Config.RABBITMQ_HTTP_PORT}/api/connections/",
+                            auth=HTTPBasicAuth(Config.RABBITMQ_USER, Config.RABBITMQ_PASSWORD))
+
+    all_connections = response.json()
+
+    node_data = {}
+
+    for connection_details in all_connections:
+
+        node_name = connection_details.get('node')
+        user_name = connection_details.get('user')
+        num_of_channels = connection_details.get('channels')
+
+        user_details = None
+        node_details = node_data.get(node_name)
+
+        if node_details:
+            user_details = node_details.get(user_name)
+        else:
+            node_data[node_name] = {}
+
+        if not user_details:
+            user_details = {
+                'number_of_connections': 0,
+                'number_of_channels': 0
+            }
+
+            node_data[node_name][user_name] = user_details
+
+        user_details['number_of_connections'] = user_details['number_of_connections'] + 1
```

Comment:
Why add 1 here?

Reply:
if there are 10 RESPONDENT-HOME connections, for example, they will each have an entry in the array of connections. This is just saying "increment the number of connections that I have seen with the same username".